Assets/MRTK/SDK/Features/UX/Scripts/InteractiveElement/States/StateManager.cs

```diff
@@ -0,0 +1,422 @@
+        public InteractionState SetStateOn(CoreInteractionState coreState)
+        {
+            return SetStateOn(coreState.ToString());
+        }
+
+        /// <summary>
+        /// Gets and sets a state to Off and invokes the OnStateDeactivated event.  Setting a 
+        /// state off changes the state value to 0.
+        /// </summary>
+        /// <param name="stateName">The name of the state to set to off</param>
+        /// <returns>The state that was set to off</returns>
+        public InteractionState SetStateOff(string stateName)
+        {
+            InteractionState state = GetState(stateName);
+
+            if (state != null)
+            {
+                if (state.Value != 0)
+                {
+                    state.Value = 0;
+                    state.Active = 0;
+
+                    // If the only state in active states is going to be removed, then activate the default state
+                    if (activeStates.Count == 1 && activeStates.First() == state)
+                    {
+                        SetStateOn(CoreInteractionState.Default);
+                    }
+
+                    // We need to save the last state active state so we can add transitions 
+                    OnStateDeactivated.Invoke(state, activeStates.Last());
+
+                    activeStates.Remove(state);
+                }  
+            }
+            else
+            {
+                Debug.LogError($"The {stateName} state is not being tracked, add this state using AddState(state) to set it");
+            }
+
+            return state;
+        }
+
+        /// <summary>
+        /// Gets and sets a Core Interaction State to Off and invokes the OnStateDeactivated event.  Setting a 
+        /// state off changes the state value to 0.
+        /// </summary>
+        /// <param name="coreState">The Core Interaction state to set to off</param>
+        /// <returns>The Core Interaction state that was set to off</returns>
+        public InteractionState SetStateOff(CoreInteractionState coreState)
+        {
+            return SetStateOff(coreState.ToString());
+        }
+
+        /// <summary>
+        /// Adds a new Core Interaction State to track.
+        /// </summary>
+        /// <param name="coreState">The Core Interaction State to add</param>
+        /// <returns>The newly added Core Interaction State</returns>
+        public InteractionState AddCoreState(CoreInteractionState coreState)
+        {
+            InteractionState state = GetState(coreState);
+
+            if (state == null)
+            {
+                // Create a new core state
+                InteractionState newState = new InteractionState(coreState.ToString());
+
+                states.Add(newState);
+
+                // Set the event configuration if one exists for the core interaction state
+                EventReceiverManager.SetEventConfiguration(newState);
```

Comment:
It seems this method has the side effect of also "Setting an event configuration" -- mention this in comment that this is a side effect.